tools/toolchains/arm.py

```diff
@@ -70,7 +72,13 @@ def __init__(self, target, notify=None, macros=None,
             raise NotSupportedException(
                 "this compiler does not support the core %s" % target.core)
 
-        self.check_c_lib_supported(target, "arm")
+        toolchain = "arm"
+
+        if is_std_c_lib_supported_instead_small_c_lib(target, toolchain):
+            # Replace small C lib with std if there is no small C lib support
+            target.c_lib = "std"
+
```

Comment:
This looks like code common to all toolchains. Can it be moved to the base class and used only once?

Reply:
Please review 0e9c83e15985570a1b18cea3e794d8f92c9d9ee1